### PR TITLE
cast json.Number

### PR DESCRIPTION
**Pull request by yanmhlv**
- [x] cast Int, Int64, Int32, Int16 and Int8
- [x] cast Uint, Uint64, Uint32, Uint16 and Uint8
- [x] cast Float64 and Float32

I think it could be useful to check float/int/uint overflow, What do you think?

### CODE REVIEW

File: caste.go [hunk 1]
```diff
@@ -479,6 +534,12 @@ func ToUintE(i interface{}) (uint, error) {
 		return 0, nil
 	case nil:
 		return 0, nil
+	case fmt.Stringer:
+		v, err := strconv.ParseUint(s.String(), 10, 64)
```
Comment (bep) [inline caste.go]: Is this relevant to this particular PR?

File: types.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package cast
+
+type (
+	Float64er interface {
```
Comment (bep) [inline types.go]: No need to export these (as then they would have to have some kind of documentation)

File: caste.go (hunk 1 above)
Comment (yanmhlv) [reply caste.go]: `ParseUint` is relevant
`fmt.Stringer`? no, I changed it on `json.Number`

**Comment by yanmhlv** [comment]
@bep, I fixed. Could you like to approve PR?

**Comment by yanmhlv** [comment]
Hello @bep ! Is my PR ok?
